Tasks/Maven/maventask.ts

```diff
@@ -227,14 +227,43 @@ function applyPmdGoals(mvnRun: trm.ToolRunner):void {
 }
 
 // Returns true if any one code analysis tool was enabled.
-function isCodeAnalysisReportRequired():boolean {
+function isCodeAnalysisUploadRequired():boolean {
     return tl.getInput('pmdAnalysisEnabled', true) == 'true';
 }
 
+// Take a result object and attempt to upload the XML and HTML reports as artifacts. Throws if fail.
+function uploadBuildArtifactsFromResult(analysisResult:ar.AnalysisResult):void {
+    // make sure the staging folder exists and is empty
+    tl.rmRF(stagingDir);
+    tl.mkdirP(stagingDir);
+
+    var filesToUpload:string[] = [];
+    if (analysisResult.xmlFilePath) {
+        filesToUpload.push(analysisResult.xmlFilePath)
+    }
+    if (analysisResult.htmlFilePath) {
+        filesToUpload.push(analysisResult.htmlFilePath);
+    }
+
+    filesToUpload.forEach(fileToUpload => {
+        // Need copy files to the staging folder and upload the entire folder for compatibility with the Xplat agent
```

Comment:
This is actually a temporary workaround to overcome a bug, not a compatibility issue. Please rephrase the comment and - since the bug is public - add a link to it.